src/Symfony/Component/HttpKernel/Attribute/QueryParameter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+/**
+ * Can be used in a controller action to mark a parameter as a query parameter.
+ *
+ * public function homeAction(
+ *   #[QueryParameter] ?string $name
+ * );
+ *
+ * When opening this URL: /home?name=John
+ * It will fill the $name parameter with the value of the query parameter "name".
+ *
+ * When opening this URL: /home
+ * It will set $name to null.
+ *
+ * When the parameter is non-nullable / not optional:
+ * public function homeAction(
+ *   #[QueryParameter] string $name
+ * );
+ *
+ * Opening /home without passing ?name=John will throw a Bad Request exception.
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class QueryParameter
```

Comment:
Should be named MapQueryParameter to build on the existing convention

Reply:
Fixed.